### PR TITLE
Test/decision_maker

### PR DESCRIPTION
**Pull request by Ravleen-Solulab**
This is the PR for decision_maker_abci.
Covering : states and rounds, dialogues.

### CODE REVIEW

**Comment by Ravleen-Solulab** [comment]
This PR contains the test script for the states(14 files), rounds and dialogues of decision_maker_abci.
Update: Everything is completed except the missing coverage in the test_redeem.py.

File: poetry.lock [hunk 1]
```diff
@@ -3581,4 +3811,4 @@ type = ["pytest-mypy"]
 [metadata]
 lock-version = "2.0"
 python-versions = "<3.12,>=3.8"
-content-hash = "d6ff2aadbc8e17ec0cf3d58f0f72d3896302520325f5f69aa2ad78e530b38a96"
+content-hash = "43c654a308c15072bda54f5c78ee81e355a9c7172183b4deeb3e61f2728eb6f8"
```
Comment (annasambrook) [inline poetry.lock:3814]: How come so many version updates to the packages here?

File: packages/valory/skills/decision_maker_abci/tests/states/test_blacklising.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    with patch.object(
+        UpdateBetsRound, "end_block", return_value=(synced_data, Event.DONE)
+    ) as mock_super_end_block:
+        result = blacklisting_round.end_block()
+
+        mock_super_end_block.assert_called_once()
+        assert result == (
+            synced_data,
+            Event.MOCK_TX,
+        )  # Should return MOCK_TX since benchmarking is enabled
```
Comment (annasambrook) [inline packages/valory/skills/decision_maker_abci/tests/states/test_blacklising.py:106]: can these be parametrized?

**Review by annasambrook — COMMENTED** [review]
Left a couple of comments, otherwise LGTM

**Review by annasambrook — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -66,7 +66,7 @@ code-checks:
 .PHONY: security
 security:
 	tomte check-security
-	gitleaks detect --report-format json --report-path leak_report
```
Comment (Adamantios) [inline Makefile:69]: Please revert.

**Review by Adamantios — CHANGES_REQUESTED** [review]
There is still some uncovered code in rounds, but we can merge on `main` and continue from there.

**Review by Adamantios — APPROVED** [review]

**Pull request merged**